### PR TITLE
Support dropdown without link

### PR DESCRIPTION
**Pull request by gregberge**
CleanShot 2024-12-03 at 16 32 52@2x

### CODE REVIEW

File: packages/gitbook/src/components/Header/HeaderLink.tsx [hunk 1]
```diff
@@ -52,20 +65,20 @@ export async function HeaderLink(props: {
...
                 return (
                     <Button
-                        href={target.href}
+                        href={target?.href}
                         variant={variant}
                         size="medium"
                         className={tcls(
                             {
-                                'button-primary':
+                                ['button-primary']:
```
Comment (scazan) [inline packages/gitbook/src/components/Header/HeaderLink.tsx:73]: just curious, why the change with the brackets?

File: packages/gitbook/src/components/Header/HeaderLink.tsx [hunk 2]
```diff
@@ -52,20 +65,20 @@ export async function HeaderLink(props: {
                 })();
                 return (
                     <Button
-                        href={target.href}
+                        href={target?.href}
```
Comment (scazan) [inline packages/gitbook/src/components/Header/HeaderLink.tsx:68]: Can this result in a "dead button" since we don't pass an onClick? should we check for it?

**Review by scazan — APPROVED** [review]
left a comment but looks good.

File: packages/gitbook/src/components/Header/HeaderLink.tsx (hunk 1 above)
Comment (gregberge) [reply packages/gitbook/src/components/Header/HeaderLink.tsx:73]: It was just a change and another change.

File: packages/gitbook/src/components/Header/HeaderLink.tsx (hunk 2 above)
Comment (gregberge) [reply packages/gitbook/src/components/Header/HeaderLink.tsx]: I simplified the whole logic @scazan

Comment (scazan) [reply packages/gitbook/src/components/Header/HeaderLink.tsx]: looks great 👍
